odometry_conversion/src/OdometryConversion.cpp

```diff
@@ -9,20 +9,23 @@ OdometryConversion::OdometryConversion(ros::NodeHandle& nh) : buffer_(), transfo
   outOdomFrame_ = nh.param<std::string>("out_odom_frame", outOdomFrame_);
   inSensorFrame_ = nh.param<std::string>("in_sensor_frame", inSensorFrame_);
   outSensorFrame_ = nh.param<std::string>("out_sensor_frame", outSensorFrame_);
+  odomChild_ = nh.param<bool>("is_odom_child", odomChild_);
 
   auto sensorTransform = buffer_.lookupTransform(outSensorFrame_, inSensorFrame_, ros::Time(0), ros::Duration(10));
   sensorTransformHom_ = toHomTransform(sensorTransform.transform);
-  sensorTransformHom_.block<3, 3>(0, 0) = sensorTransformHom_.block<3, 3>(0, 0).transpose().eval();
-  sensorTransformHom_.block<3, 1>(0, 3) = -sensorTransformHom_.block<3, 3>(0, 0) * sensorTransformHom_.block<3, 1>(0, 3);
 
-  auto odomTransform = buffer_.lookupTransform(outSensorFrame_, inOdomFrame_, ros::Time(0), ros::Duration(10));
+  // if frame exists, otherwise make a new frame at the same place
+  auto odomTransform = buffer_.lookupTransform(outSensorFrame_, outSensorFrame_, ros::Time(0), ros::Duration(10));  // identity
+  if (buffer_.canTransform(outOdomFrame_, inOdomFrame_, ros::Time(0), ros::Duration(10))) {
+      odomTransform = buffer_.lookupTransform(outOdomFrame_, inOdomFrame_, ros::Time(0), ros::Duration(10));
+  }
   odomTransformHom_ = toHomTransform(odomTransform.transform);
 
   // the odom frame of the tracking camera is gravity aligned. In case the camera is mounted with an angle, remove pitch and roll of this
   // transformation
-  odomTransform.header.frame_id = outOdomFrame_;
-  odomTransform.child_frame_id = inOdomFrame_ + "_odom";
-  odomCameraOdomTransformPublisher_.sendTransform(odomTransform);
+  // odomTransform.header.frame_id = outOdomFrame_;
+  // odomTransform.child_frame_id = inOdomFrame_ + "_odom";
+  // odomCameraOdomTransformPublisher_.sendTransform(odomTransform);
 
   odometryPublisher_ = nh.advertise<nav_msgs::Odometry>("/base_odom", 1, false);
   odometryInSubscriber_ = nh.subscribe("/tracking_camera/odom/sample", 1, &OdometryConversion::odometryInCallback, this);
```

Comment:
Could we make those two topic names a parameter as well, while we're making changes to the odometry_conversion node?  that would make it more flexible.

Reply:
Done